src/main/java/de/zalando/tip/zalenium/util/GoogleAnalyticsApi.java

```diff
@@ -0,0 +1,92 @@
+package de.zalando.tip.zalenium.util;
+
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ByteArrayEntity;
+import org.apache.http.impl.client.HttpClientBuilder;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Small class to create hits in Google Analytics
+ * We want to track when a test starts and finishes, also information about errors (Exceptions)
+ */
+public class GoogleAnalyticsApi {
+
+    private final Logger LOGGER = Logger.getLogger(GoogleAnalyticsApi.class.getName());
+
+    private HttpClient httpClient;
+
+    private static final String GA_TRACKING_ID = "GA_TRACKING_ID";
+    private static final String GA_API_VERSION = "GA_API_VERSION";
+    private static final String GA_ANONYMOUS_CLIENT_ID = "GA_ANONYMOUS_CLIENT_ID";
+
+    private final Environment defaultEnvironment = new Environment();
+    private Environment env = defaultEnvironment;
+
+    public GoogleAnalyticsApi() {
+        this.httpClient = HttpClientBuilder.create().build();
+    }
+
+    @VisibleForTesting
+    protected void setHttpClient(HttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    @VisibleForTesting
+    protected void setEnv(final Environment env) {
+        this.env = env;
+    }
+
+    public void startTestEvent(String nodeHandler, String capabilities) {
+        trackEvent("test_start", nodeHandler, capabilities);
+    }
+
+    public void stopTestEvent(String nodeHandler) {
+        trackEvent("test_stop", nodeHandler, "");
+    }
+
+    public void trackException(Exception e) {
+        String payload = String.format("v=%s&tid=%s&cid=%s&t=exception&exd=%s&exf=%s", GA_API_VERSION, GA_TRACKING_ID,
+                GA_ANONYMOUS_CLIENT_ID, e.getMessage(), "0");
+        doPost(payload);
+    }
+
+    private void trackEvent(String eventCategory, String eventAction, String eventLabel) {
+        String payload = String.format("v=%s&tid=%s&cid=%s&t=event&ec=%s&ea=%s&el=%s", GA_API_VERSION,
+                GA_TRACKING_ID, GA_ANONYMOUS_CLIENT_ID, eventCategory, eventAction, eventLabel);
+        doPost(payload);
+    }
+
+
+    private void doPost(String payload) {
+        new Thread(() -> {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this lambda expression number of lines from 21 to at most 20. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1188)